plug.vim

```diff
@@ -1208,7 +1208,7 @@ function! s:spawn(name, cmd, opts)
   let cmd = has_key(a:opts, 'dir') ? s:with_cd(a:cmd, a:opts.dir) : a:cmd
   if !empty(job.batchfile)
     call writefile(["@echo off\r", cmd . "\r"], job.batchfile)
-    let cmd = s:shellesc(expand(job.batchfile))
+    let cmd = s:shellesc(expand(job.batchfile), &shell)
```

Comment:
@mikelue Does `bash -c 'C:/foo/bar.bat'` (with or without -c) work? If not, `expand` doesn't cut it. Is `cygpath.exe` in `$PATH`?

Reply:
I was silly. I didn't get the patch on my computer.

Nevertheless,  thing seems working for `bash -c 'C:/foo/bar.bat'`.

```bat
" for validate
@echo off
cd /d ^"C:\Users\mike\vimfiles\plugged\delimitMate\\^" && git rev-parse --abbrev-ref HEAD 2>&1 && git config -f .git/config remote.origin.url

" for fetch
@echo off
cd /d ^"C:\Users\mike\vimfiles\plugged\delimitMate\\^" && git fetch   2>&1
```

But the result still got a little wired:
```
Updated. Elapsed time: 1.497148 sec.
[x]

- Finishing ... Done!
x delimitMate:
    OK
```
Expected one:
```
 Updated. Elapsed time: 1.343617 sec.     
[=]                                                                                

 - Finishing ... Done!                                                                  
 - delimitMate: Already up to date.                                                 
```

I try to print something in `s:spawn()` at `line: 1232` of `argv`
```
cmd /c 'C:/Users/mike/AppData/Local/Temp/VIE9C7.tmp.bat'
```

printing of `s:job_exit_cb(self, data)`
```
a:self >> {'jobid': 'process 26788 dead', 'name': 'delimitMate', 'new': 0, 'batchfile': 'C:/Users/mike/AppData/Local/Temp/VIE9C7.tmp.bat', 'lines': [''], 'running': 1, 'error': 0}
a:data >> 1
```

I have no idea what's going on. Thanks again.

Reply:
`cmd /c` shouldn't have happened. Thanks for debugging.